### PR TITLE
Recipe App Completion

### PR DESCRIPTION
**Pull request by patriciachrysy**
Dear reviewer, thank you for taking time to review our project. Here are some guidelines to what has been done: 

- should follow the layout of the wireframes provided. You should personalize the rest of the design including colors, typographies, spacings, etc.

Login page and registration page:
- Should be built with Devise.

Recipes list:
- Should display a list of recipes created by the logged-in user as in the wireframe.
- Should lead to recipe details.

Public recipe list:
- Should display a list of all public recipes ordered by newest as in the wireframe.
- Should lead to recipe details.
- If the user is the owner of the recipe, should allow the user to delete it.

Recipe details:
- Should display a toggle button that allows for a recipe to be made public or private.
- If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
- If the user is the owner of the recipe, should lead to the form that allows the user to add new food.
- Make sure there are no N+1 queries happening.

- Create a navigation menu that allows users to open all of the pages you created.

- Write unit and integration tests

Food list:
- Should display a list of food added by the logged-in user as in the wireframe (display also quantity of a given food).
- Should lead to a form that allows users to add new food.
General shopping list view:
- Should show the list of food that is missing for all recipes of the logged-in user (compare the list of food for all recipes with the general food list of that user).
- Should count the total food items and total price of the missing food.

### CODE REVIEW

File: app/views/recipes/_recipe.html.erb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        </div>
+        <div class='recipe-info'>
+        <div class='recipe-n'>
+            <%= link_to "#{recipe.name}", recipe_path(recipe), class: 'recipe-name' %>
+            <%= link_to "Remove", recipe_path(recipe), data: {
+                    turbo_method: :delete,
+                    turbo_confirm: 'Are you sure you want to delete this recipe?'
+                }, class: 'remove' if can? :destroy, recipe %>
+        </div>
+        <div>
```
Comment (LionRouge1) [inline app/views/recipes/_recipe.html.erb:13]: - [ ] When trying to delete a recipe that have foods I got **InvalidForeignKey** error (see image).
Screenshot from 2023-12-23 18-07-49
This error can be fixed by adding `dependent: :destroy` to the the recipe and recipe_foods in the recipe model
```rb
    has_many :recipe_foods, dependent: :destroy
```

File: app/views/foods/_food.html.erb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    <td>$<%= food.price %></td>
+    <% if with_actions %>
+        <td class='action-buttons'>
+            <%= link_to "Modify", edit_food_path(food), class: 'info-bg' if can? :update, food %>
+            <%= link_to "Delete", food_path(food), data: {
+                    turbo_method: :delete,
+                    turbo_confirm: 'Are you sure you want to delete this food?'
+                }, class: 'danger-bg' if can? :destroy, food %>
+        </td>
+    <% end %>
```
Comment (LionRouge1) [inline app/views/foods/_food.html.erb:13]: - [ ] When trying to delete a Food that have recipes I got **InvalidForeignKey** error (see image).
Screenshot from 2023-12-23 18-07-49
This error can be fixed by adding `dependent: :destroy` to the the Food and recipe_foods in the recipe model
```rb
    has_many :recipe_foods, dependent: :destroy
```

**Review by LionRouge1 — CHANGES_REQUESTED** [review]
Hi **Team**, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- The App has a login and Registration page, built with the Devise gem :heavy_check_mark: 
- All the written tests ( Unit and Integration) are parsed :heavy_check_mark: 
- The application run without any error :x:

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/recipes/_recipe.html.erb (hunk 1 above)
Comment (Abenezer-Tilahun) [reply app/views/recipes/_recipe.html.erb:13]: Fixed. Successfully delete the recipe.
image

File: app/views/foods/_food.html.erb (hunk 2 above)
Comment (Abenezer-Tilahun) [reply app/views/foods/_food.html.erb:13]: Fixed. Successfully deleted Food.
image

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @patriciachrysy @zhorabay 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes. 
✔️ The application run as expected and All test cases are passing.
image

✔️ Good job for Implemented RoR group project: day 3.


## Optional suggestions 


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

**Pull request merged**